compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/InputObserver.skiko.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.platform
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+
+@kotlin.native.concurrent.ThreadLocal
+private val textInputListeners = mutableSetOf<(String) -> Unit>()
```

Comment:
Here I used global ThreadLocal state
```val textInputListeners = mutableSetOf```

Reply:
I'm curious to learn why is it necessary to add ThreadLocal here?

Reply:
I was trying to make a local val textInputListeners = mutableSetOf(). But it fails with exception BadAddres access or something similar

Reply:
I am just tryed to remove @ThreadLocal, and stay global val textInputListeners. It work's good. And because I modify this MutableList only inside MainScope(), - then no need to @ThreadLocal.
I will remove ThreadLocal.

Reply:
Was it previously modified from different scopes when it failed?

Reply:
Previously: When inside InputObserver class as local **val**, then it fails with exception.

Reply:
> Was it previously modified from different scopes when it failed?

Previously it also modifies from MainScope() only